ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,461 @@
+        let raw_mustache_details: [bool; 16] = get_and_concat_vec_bit(&raw_mii_data, &[0x40, 0x41])
+            .try_into()
+            .unwrap();
+        let raw_beard_details: [bool; 16] = get_and_concat_vec_bit(&raw_mii_data, &[0x42, 0x42])
+            .try_into()
+            .unwrap();
+        let raw_glass_details: [bool; 16] = get_and_concat_vec_bit(&raw_mii_data, &[0x44, 0x45])
+            .try_into()
+            .unwrap();
+        let raw_mole_details: [bool; 16] = get_and_concat_vec_bit(&raw_mii_data, &[0x46, 0x47])
+            .try_into()
+            .unwrap();
+        let raw_utf16_author = &raw_mii_data[0x48..0x5C];
+
+        let mii_name = utf16_byte_pairs_to_string(raw_utf16_name);
+        let author_name = utf16_byte_pairs_to_string(raw_utf16_author);
+
+        let options = MiiDataOptions {
+            is_copying_allowed: raw_options[0],
+            is_profanity_flag_enabled: raw_options[1],
+            region_lock: {
+                match (raw_options[3], raw_options[2]) {
+                    (false, false) => RegionLock::None,
+                    (false, true) => RegionLock::Japan,
+                    (true, false) => RegionLock::USA,
+                    (true, true) => RegionLock::Europe,
+                }
+            },
+            charset: {
+                match (raw_options[5], raw_options[4]) {
+                    (false, false) => Charset::JapanUSAEurope,
+                    (false, true) => Charset::China,
+                    (true, false) => Charset::Korea,
+                    (true, true) => Charset::Taiwan,
+                }
+            },
+        };
+
+        let position = SelectorPosition {
+            page_index: partial_u8_bits_to_u8(&raw_position[0..3]),
+            slot_index: partial_u8_bits_to_u8(&raw_position[4..7]),
+        };
+
+        let device = ConsoleIdentity {
+            origin_console: {
+                match (raw_device[6], raw_device[5], raw_device[4]) {
+                    (false, false, true) => OriginConsole::ConsoleWii,
+                    (false, true, false) => OriginConsole::ConsoleDSi,
+                    (false, true, true) => OriginConsole::Console3DS,
+                    _ => OriginConsole::ConsoleWiiUSwitch,
+                }
+            },
+        };
+
+        let details = Details {
+            sex: {
+                match raw_details[0] {
+                    true => MiiSex::Female,
+                    false => MiiSex::Male,
+                }
+            },
+            birthday_month: partial_u8_bits_to_u8(&raw_details[1..4]),
+            birthday_day: partial_u8_bits_to_u8(&raw_details[5..9]),
+            shirt_color: partial_u8_bits_to_u8(&raw_details[10..13]),
+            is_favorite: raw_details[14],
+        };
+
+        let face_details = FaceDetails {
+            style: FaceStyle {
+                is_sharing_enabled: !raw_face_style[1],
+                shape: partial_u8_bits_to_u8(&raw_face_style[1..4]),
```

Comment:
Are you sure `is_sharing_enabled` uses `raw_face_style[1]`? `shape` also uses that value.